curator-framework/src/main/java/org/apache/curator/framework/imps/CreateBuilderImpl.java

```diff
@@ -485,7 +485,6 @@ private String protectedPathInForeground(String adjustedPath, byte[] data) throw
         }
         catch ( Exception e)
         {
-            ThreadUtils.checkInterrupted(e);
```

Comment:
I've just pushed a basic test case showing the issue. The thread is interrupted once, but it is exposed to the caller twice - once through close() throwing an exception, and once through whatever next checks the interrupt status (here, I've used Thread.sleep)